Bigtable/src/ChunkFormatter.php

```diff
@@ -136,13 +176,60 @@ public function readAll()
                     $rowKey = $this->rowKey;
                     $this->commit();
                     yield $rowKey => $row;
+                    $this->numberOfRowsRead++;
                 }
             }
         }
 
         $this->onStreamEnd();
     }
 
+    public function retryingArgumentProvider()
+    {
+        $prevRowKey = $this->prevRowKey;
+        $this->reset();
+        if ($prevRowKey) {
+            $this->updateOptions($prevRowKey);
+        }
+        return [$this->tableName, $this->options];
+    }
+
+    private function updateOptions($prevRowKey)
+    {
+        if ($this->originalRowsLimit) {
+            $this->options['rowsLimit'] = $this->originalRowsLimit - $this->numberOfRowsRead;
+        }
+        if (isset($this->options['rows'])) {
+            $rowSet = $this->options['rows'];
+            if (count($rowSet->getRowKeys()) > 0) {
+                $rowKeys = [];
+                foreach ($rowSet->getRowKeys() as $rowKey) {
+                    if ($rowKey > $prevRowKey) {
```

Comment:
Just a quick question to confirm here, does PHP's internal behavior match the sorting order in Bigtable?

Reply:
yes, Bigtable row keys are sorted lexicographically